BMS_Sender/BMS_Sender.cpp

```diff
@@ -0,0 +1,62 @@
+/* ***********************************************************************************************************
+* File Name   :	BMS_Sender.cpp
+* Author      : Arun Raveendran
+* Description : BMS_Sender.cpp is used to send the BMS datas like temperature and State of Charging
+* Functions	  :
+* *********************************************************************************************************** */
+#include "BMS_Sender.h"
+#include <ctime>
+
+CBMSDataGenerator:: CBMSDataGenerator()
+{
+    srand((unsigned int)time(NULL));
+}
+
+bool CBMSDataGenerator::ValidateDataRange(float value, int minVal, int maxVal)
+{
+    return((value >= minVal) && (value <= maxVal));
+}
+
+int CBMSDataGenerator::RandomNumberGenerator(int min, int max)
+{
+    return (rand() % ((max - min) + 1)) + min;
+}
+
+float CBMSDataGenerator::GenerateStateOfCharge()
+{
+    return RandomNumberGenerator(MIN_SOC, MAX_SOC);
+}
+
+float CBMSDataGenerator::GenerateTemperature()
+{
+    return RandomNumberGenerator(MIN_TEMPERATURE, MAX_TEMPERATURE);
+}
+
+std::stringstream CBMSDataGenerator::GenerateJSON()
+{
+    stringstream data;
+
+    m_param.SetTemperature(GenerateTemperature());
+    m_param.SetStateOfCharge(GenerateStateOfCharge());
+    data << "{\"Temperature\": "<< m_param.GetTemperature() << " ,\"SOC\": " << m_param.GetStateOfCharge() <<" }";
```

Comment:
`CBMSParam` is a class with only set and get methods. Usually, this indicates an opportunity for more encapsulation. E.g., the above formatting can go into that class itself. Then, any change to add parameters is 'encapsulated' in `CBMSParam` itself.